src/components/Frame/Frame.tsx

```diff
@@ -66,340 +56,292 @@ const APP_FRAME_NAV = 'AppFrameNav';
-        intl,
-        mediaQuery: {isNavigationCollapsed},
-      },
-    } = this.props;
-
-    const navClassName = classNames(
-      styles.Navigation,
-      showMobileNavigation && styles['Navigation-visible'],
-    );
+  }, [skipToContentTarget]);
 
-    const mobileNavHidden = isNavigationCollapsed && !showMobileNavigation;
-    const mobileNavShowing = isNavigationCollapsed && showMobileNavigation;
-    const tabIndex = mobileNavShowing ? 0 : -1;
-
-    const navigationMarkup = navigation ? (
-      <TrapFocus trapping={mobileNavShowing}>
-        <CSSTransition
-          findDOMNode={this.findNavigationNode}
-          appear={isNavigationCollapsed}
-          exit={isNavigationCollapsed}
-          in={showMobileNavigation}
-          timeout={durationSlow}
-          classNames={navTransitionClasses}
-        >
-          <div
-            ref={this.navigationNode}
-            className={navClassName}
-            onKeyDown={this.handleNavKeydown}
-            id={APP_FRAME_NAV}
-            key="NavContent"
-            hidden={mobileNavHidden}
-          >
-            {navigation}
-            <button
-              type="button"
-              className={styles.NavigationDismiss}
-              onClick={this.handleNavigationDismiss}
-              aria-hidden={
-                mobileNavHidden ||
-                (!isNavigationCollapsed && !showMobileNavigation)
-              }
-              aria-label={intl.translate(
-                'Polaris.Frame.Navigation.closeMobileNavigationLabel',
-              )}
-              tabIndex={tabIndex}
-            >
-              <Icon source={MobileCancelMajorMonotone} color="white" />
-            </button>
-          </div>
-        </CSSTransition>
-      </TrapFocus>
-    ) : null;
+  const showToast = useCallback((toast: ToastPropsWithID) => {
+    setToastMessages((toastMessages) => {
+      const hasToastById = Boolean(
+        toastMessages.find(({id}) => id === toast.id),
+      );
 
-    const loadingMarkup =
-      loadingStack > 0 ? (
-        <div className={styles.LoadingBar} id={APP_FRAME_LOADING_BAR}>
-          <Loading />
-        </div>
-      ) : null;
+      return hasToastById ? toastMessages : [...toastMessages, toast];
+    });
+  }, []);
 
-    const contextualSaveBarMarkup = (
-      <CSSAnimation
-        in={showContextualSaveBar}
-        className={styles.ContextualSaveBar}
-        type={AnimationType.Fade}
-      >
-        <ContextualSaveBar {...this.contextualSaveBar} />
-      </CSSAnimation>
+  const hideToast = useCallback(({id}: ToastID) => {
+    setToastMessages((toastMessages) =>
+      toastMessages.filter(({id: toastId}) => id !== toastId),
     );
-
-    const topBarMarkup = topBar ? (
-      <div
-        className={styles.TopBar}
-        {...layer.props}
-        {...dataPolarisTopBar.props}
-        id={APP_FRAME_TOP_BAR}
+  }, []);
+
+  const startLoading = useCallback(() => {
+    setLoadingStack((loadingStack) => loadingStack + 1);
```

Comment:
Loving the use of the callback style to avoid needing to depend upon the the current value of loadingStack! Nice one!